python/cugraph/cugraph/link_prediction/jaccard.py

```diff
@@ -108,55 +142,75 @@ def jaccard(input_graph, vertex_pair=None, do_expensive_check=True):
     Examples
     --------
     >>> from cugraph.datasets import karate
-    >>> G = karate.get_graph(download=True)
-    >>> df = cugraph.jaccard(G)
+    >>> from cugraph import jaccard
+    >>> input_graph = karate.get_graph(download=True, ignore_weights=True)
+    >>> df = jaccard(input_graph)
 
     """
     if do_expensive_check:
-        if not input_graph.renumbered:
-            input_df = input_graph.edgelist.edgelist_df[["src", "dst"]]
-            max_vertex = input_df.max().max()
-            expected_nodes = cudf.Series(range(0, max_vertex + 1, 1)).astype(
-                input_df.dtypes[0]
-            )
-            nodes = (
-                cudf.concat([input_df["src"], input_df["dst"]])
-                .unique()
-                .sort_values()
-                .reset_index(drop=True)
-            )
-            if not expected_nodes.equals(nodes):
-                raise ValueError("Unrenumbered vertices are not supported.")
+        warnings.warn(
+            "do_expensive_check is deprecated since it is no longer needed",
+            DeprecationWarning,
```

Comment:
For this and others, we're raising `FutureWarning` now instead, as documented [here](https://docs.rapids.ai/api/cudf/nightly/developer_guide/contributing_guide/#deprecating-and-removing-code).

We're also planning to update the utils that raise `DeprecationWarning` in a separate PR, but feel free to do that here too if it's not too distracting to the PR.